lib/xlibc/src/threads/_xlibc_thread_startup.c

```diff
@@ -1,8 +1,31 @@
 #include <threads.h>
+#include <assert.h>
+#include <string.h>
 #include "xboxkrnl/xboxkrnl.h"
+#include <_tls.h>
 
 VOID NTAPI _xlibc_thread_startup (PKSTART_ROUTINE StartRoutine, PVOID StartContext)
 {
+    // Make sure the TLS pointer is valid
+    assert(KeGetCurrentThread()->TlsData);
+
+    // Retrieve TLS area address and point first entry to itself
+    DWORD *TlsData;
+    TlsData = ((DWORD *)KeGetCurrentThread()->TlsData) + 1;
+    TlsData[-1] = (DWORD)TlsData;
+
+    // 16 byte alignment required
+    assert(((DWORD)TlsData & 15) == 0);
+
+    // copy initial values of TLS area
+    DWORD TlsDataSize;
+    TlsDataSize = _tls_used.EndAddressOfRawData - _tls_used.StartAddressOfRawData;
+    memcpy(TlsData, (void *)_tls_used.StartAddressOfRawData, TlsDataSize);
+
+    // zero-initialize the rest
+    if (_tls_used.SizeOfZeroFill != 0)
+        RtlZeroMemory((char *)TlsData + TlsDataSize, _tls_used.SizeOfZeroFill);
```

Comment:
Explicit scope please.

Not sure if we do it all over the codebase yet, but *I* think we should.

Reply:
Will do.